docs/user_guide/README.md

```diff
@@ -162,25 +162,107 @@ Once all parts have been uploaded, the user must call the provided complete uplo
+    --header "Content-Type: application/json" \
+    --header "Authorization: Bearer $ACCESS_TOKEN" \
+    --data "{
+        \"path\":\"${DESTINATION_PATH}\",
+        \"account\": \"${ACCOUNT}\",
+        \"fileName\":\"${LARGE_FILE_NAME}\",
+        \"fileSize\":\"${LARGE_FILE_SIZE_IN_BYTES}\"
+    }"
+    ```
+
+The JSON response from this call follows the structure shown below. FirecREST calculates the number of parts the file must be split into, based on the provided file size and the `maxPartSize` setting. Each part is assigned a number from <i>1</i> to <i>n</i> and must be uploaded using the presigned URLs listed in `partsUploadUrls`. Once all parts have been successfully uploaded, the presigned URL in `completeUploadUrl` is used to finalize the upload sequence and initiate the transfer of the complete data file from S3 to its final destination.
+
+!!! example "FirecREST response from `/filesystem/{system}/transfer/upload` endpoint"
+    ```json
+    {
+    "transferJob": {
+        "jobId": nnnnnnnnn,
+        "system": "SYSTEM",
+        "workingDirectory": "/xxxxxxxxx",
+        "logs": {
+            "outputLog": "/xxxxxxxx.log",
+            "errorLog": "/xxxxxxxxx.log"
+        }
+    },
+    "partsUploadUrls": [
+        "https://part1-url", "https://part2-url", "https://part3-url"
+    ],
+    "completeUploadUrl": "https://upload-complete-url",
+    "maxPartSize": 1073741824
+    }
+    ```
+Given the `maxPartSize` field in the `/filesystem/{system}/transfer/upload` end-point response, split your large file consequently:
 
 !!! example "Split large file to upload"
     ```bash
-    $ split -n 7 -d large-file.zip large-file-part-
+    $ split "$LARGE_FILE_NAME" -b "$maxPartSize" --numeric-suffixes=1
     ```
 
-Upload each individual part following the correct part order:
+This will divide your large file in a set of parts numbered from <i>x01</i>,<i>x02</i>, etc. to the number of parts that the S3 multipart upload protocol expects. The number of split parts must match the number of items in the `partsUploadUrls`list.
+
+Upload each part in the correct order. After a successful upload, S3 responds with an `ETag`, which is a checksum of that specific part. This value is essential for completing the multipart upload, so be sure to record it.
+
+The example below demonstrates a simple sequential upload. However, this approach is not mandatory, since S3 fully supports uploading parts in parallel.
 
 !!! example "Upload parts call"
     ```bash
-    $ curl 'https://rgw.cscs.ch/firecresttds%3Auser/62ad2cd8-7398-4955-929d-cbfae5088c6a/large-file.zip?uploadId=2~qiT12y-T1Hhl_ELCozIt3ZlLhMoTcmy&partNumber=1&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=GET9Y98HGJARIS4I447Z%2F20250325%2Fcscs-zonegroup%2Fs3%2Faws4_request&X-Amz-Date=20250325T071416Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=d0edacd3fe1d3dc1e38f5d632f7760275cda29e9e41c49548b5da94e47699400' --upload-file large-file-part-00
+    part_id=1
```

Comment:
Would it make sense, as we do with `partsUploadUrls` and `completeUploadUrl`, to add the json field `uploadScript` in the response, possibly with this data already processed, to be executed by the user/client?

I understand that that would be possible only if the user splits the file in the format that we require, but we could add a "default" way of splitting the file that matches the script response.